presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -148,12 +150,40 @@ public void testPushDownJoinConditionConjunctsToInnerSideBasedOnInheritedPredica
     public void testSameScalarSubqueryIsAppliedOnlyOnce()
     {
         // three subqueries with two duplicates, only two scalar joins should be in plan
-        Plan plan = plan("SELECT * FROM orders WHERE orderkey = (SELECT 1) AND custkey = (SELECT 2) AND custkey != (SELECT 1)");
-        PlanNodeExtractor planNodeExtractor = new PlanNodeExtractor(planNode -> planNode instanceof EnforceSingleRowNode);
+        Plan plan = plan(
+                "SELECT * FROM orders WHERE orderkey = (SELECT 1) AND custkey = (SELECT 2) AND custkey != (SELECT 1)");
+        PlanNodeExtractor planNodeExtractor = new PlanNodeExtractor(EnforceSingleRowNode.class::isInstance);
         plan.getRoot().accept(planNodeExtractor, null);
         assertEquals(planNodeExtractor.getNodes().size(), 2);
     }
 
+    @Test
+    public void testSubqueryPruning()
+    {
+        List<String> subqueries = ImmutableList.of(
+                "orderkey IN (SELECT orderkey FROM lineitem WHERE orderkey % 2 = 0)",
+                "EXISTS(SELECT orderkey FROM lineitem WHERE orderkey % 2 = 0)",
+                "0 = (SELECT orderkey FROM lineitem WHERE orderkey % 2 = 0)");
+
+        for (String subquery : subqueries) {
+            // Apply can be rewritten to *join so we expect no join here as well
+            assertPlanContainsNoApplyOrJoin("SELECT COUNT(*) FROM (SELECT " + subquery + " FROM orders)");
+            // TODO uncomment below line
```

Comment:
Ok, then add a comment in the TODO like "uncomment when XXX is implemented". It wasn't clear whether that was a "to do" for the future or something you added while you were writing the code so that you'd remember to do it and then it slipped by.